docs: add info about claims

DevWithTheHair opened the pull request:
# Summary

This PR adds info about the various ways that `Claims` can be returned that describe authenticated information about the `User`.

The use case that this is trying to illustrate is how 3rd party developers can avoid receiving (and potentially _storing_) [PII](https://en.wikipedia.org/wiki/Personal_data) about the user in the form of `Claims` in the `Identity Token` yet still be able to retrieve such information on-demand when they need it (i.e. via the [UserInfo](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo) Endpoint). 

This also adds a small feature to log the API Environment to the console. This can be useful debugging purposes. The output will look something like this when folks fire up the example project locally:

```
Environment: local
API_ENVIRONMENT: https://digital.garden-fi.com
Server listening on https://localhost:8080
```

# Jira Ticket

[DX-437 Update `Consumer API OIDC Example` to use V0 auth endpoints](https://banno-jha.atlassian.net/browse/DX-437)

DevWithTheHair committed:
docs: add info about how Claims can be returned

DevWithTheHair committed:
feat: log API Environment to the console

DevWithTheHair committed:
fix: typo

samuelcouch reviewed (approved):
👍  This looks good to me.

---
**Unrelated**
This could be a different PR (or not changed at all), but since we're updating the console output, if we could remove the `...` from the listening output https://github.com/Banno/consumer-api-openid-connect-example/blob/0f6f54166ae49b94566612c96722dc9cf8dbef0d/server.js#L210-L212 it would make the URL clickable from the terminal (in iTerm at least). This is a minor change, and maybe a "me" problem, so feel free to ignore, I just noticed this again when testing 😄

DevWithTheHair committed:
feat: make console output URL clickable

Removing the ellipsis from the console output makes the URL clickable in the terminal.

DevWithTheHair commented:
@samuelcouch sure, that's a quick fix. I've gone ahead and done so in https://github.com/Banno/consumer-api-openid-connect-example/pull/23/commits/a27e9515918db80bf36483e91e99dd277a71bb6c. I also updated the Summary to have the revised terminal output example.

samuelcouch reviewed (approved):
Looks good!
